dds/DCPS/XTypes/DynamicType.h

```diff
@@ -0,0 +1,50 @@
+#ifndef OPENDDS_DCPS_XTYPES_DYNAMIC_TYPE_H
+#define OPENDDS_DCPS_XTYPES_DYNAMIC_TYPE_H
+
+#include "External.h"
+#include "DynamicTypeMember.h"
+
+#include <dds/DCPS/RcHandle_T.h>
+#include <dds/DCPS/RcObject.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+namespace OpenDDS {
+namespace XTypes {
+
+class TypeDescriptor;
+
+class OpenDDS_Dcps_Export DynamicType : public OpenDDS::DCPS::RcObject
+{
+public:
+  DynamicType();
+  ~DynamicType();
+
+  void get_descriptor(TypeDescriptor& descriptor) const;
+  DCPS::String get_name() const;
+  TypeKind get_kind() const;
+  DDS::ReturnCode_t get_member_by_name(DynamicTypeMember_rch& member, const DCPS::String& name);
+  void get_all_members_by_name(DynamicTypeMembersByName& member);
+  DDS::ReturnCode_t get_member(DynamicTypeMember_rch& member, MemberId id);
+  void get_all_members(DynamicTypeMembersById& member);
+  ACE_CDR::ULong get_member_count();
+  DDS::ReturnCode_t get_member_by_index(DynamicTypeMember_rch& member, ACE_CDR::ULong index);
+  bool equals(const DynamicType& other);
+
+  DynamicTypeMembersByName member_by_name;
+  DynamicTypeMembersById member_by_id;
+  DynamicTypeMembersByIndex member_by_index;
+  TypeDescriptor* descriptor_;
+};
+
+typedef OpenDDS::DCPS::RcHandle<DynamicType> DynamicType_rch;
+
+OpenDDS_Dcps_Export bool test_equality(const DynamicType& lhs, const DynamicType& rhs);
```

Comment:
Is this different from using the `equals` method?  Should it be the `operator==` for this type?